fabfile.py

```diff
@@ -1,197 +1,518 @@
+        unix_user=UNIX_USER
+    )
+    cron_file = "/etc/cron.d/{0}_copy".format(PROJECT_NAME)
+    local("echo '{0}' | sudo tee {1}".format(
+        output, cron_file
+    ))
+
+
+def send_error_email(error, some_env):
+    run_management_command(
+        "error_emailer '{}'".format(
+            error
+        ),
+        some_env
+    )
+
+
+@task
+def copy_backup(branch_name):
+    current_env = Env(branch_name)
+    private_settings = get_private_settings()
+    env.host_string = private_settings["host_string"]
+    env.password = private_settings["password"]
+
+    if not os.path.isfile(current_env.backup_name):
+        send_error_email(
+            "unable to find backup {}".format(
+                current_env.backup_name
+            ),
+            current_env
+        )
+    else:
+        with settings(warn_only=True):
+            failed = put(
+                local_path=current_env.backup_name,
+                remote_path=current_env.backup_name
+            ).failed
+
+        if failed:
+            send_error_email(
+                "unable to copy backup {}".format(
+                    current_env.backup_name
+                ),
+                current_env
+            )
+
+
+def get_private_settings():
+    if not os.path.isfile(PRIVATE_SETTINGS):
+        raise ValueError(
+            "unable to find additional settings at {}".format(
+                PRIVATE_SETTINGS
+            )
+        )
+
+    with open(PRIVATE_SETTINGS) as privado:
+        result = json.load(privado)
+        err_template = "we require {} in your private settings"
+        if "db_password" not in result:
+            raise ValueError(err_template.format("a db password"))
+        if "additional_settings" not in result:
+            raise ValueError(err_template.format(
+                "an additional_settings dict (even if its empty)"
             ))
-        for db_command in db_commands(username):
-            local("heroku run --app {0} {1}".format(name, db_command))
+        if "proxy" not in result:
+            raise ValueError(err_template.format("proxy variable"))
+        if "host_string" not in result:
+            e = "we host string to be set, this should be 127.0.0.1 on test, \
+or the address you want to sync to on prod in your private settings"
+            raise ValueError(e)
+
+    return result
 
 
 @task
-def production_deploy():
-    # TODO include the supervisor commands
-    # synch the nginx conf
-    with lcd(fabfile_dir):
-        local("python manage.py migrate")
-        local("python manage.py collectstatic --noinput")
+def clone_branch(branch_name):
+    branch_env = Env(branch_name)
+    local(
+        "git clone -b {0} {1} {2}".format(
+            branch_name,
+            GIT_URL,
+            branch_env.project_directory
+        )
+    )
 
 
 @task
-def checkout_project():
-    checkout(get_requirements())
+def create_private_settings():
+    with open(PRIVATE_SETTINGS, "w") as privado:
```

Comment:
If this file already exists is the right thing to do definitely to overwrite it rather than error?